idaes/core/util/tags.py

```diff
@@ -580,27 +573,20 @@ def add(self, name, expr, **kwargs):
         else:
             self[name] = ModelTag(expr=expr, **kwargs)
 
-    def table_heading(self, tags=None, units=True):
-        """Create a table heading with a given set of tags, for tabulating model
-        results.
-
-        Args:
-            tags: List of tags or tuples of tags and an index.  If a tag is for
-                an indexed value and no index is given the tag will be flattened
-                to create a column for each index.
-            units: If true include the units of measure in the coulmn heading
-                names.
-
-        Returns:
-            list of column headings
+    def _table_tagkey_index_lists(self, tags=None):
+        """PRIVATE utility function to take tags (list of tag keys or 2-element
+        list of a tag ket and index) and turn it into a list of tag keys and list
+        of indexes. Indexed tags with no index will be expanded to include all
+        indexes.
         """
+
         if tags is None:
             tags = list(self.keys())
 
         tag_list = []
         indexes = []
         for i, tag in enumerate(tags):
-            if not isinstance(tag, str):
+            if isinstance(tag, list):
```

Comment:
I'm not thrilled with this solution, because now it runs up against duck-typing, but I also haven't been able to come up with anything better. One convoluted solution would be to check whether `tag.__hash__ is None`, and if so, iterate over it to get the indexes.

Reply:
I figure there aren't many options here.  It's 99.99% going to be a tuple or a list.  I guess a tuple is very likely the only hashable list-like thing, so it seems reasonable to just say anything not hashable with a length of 2 is a key index pair.  I'll update.  I guess I'll leave the doc to say list, and maybe explicitly say tuple is not okay for a key-index pair, since it can be a key itself.